tofu/modules/harvester_host/main.tf

```diff
@@ -0,0 +1,125 @@
+resource "harvester_virtualmachine" "this" {
+  name      = "${var.project_name}-${var.name}"
+  namespace = var.namespace
+  hostname  = var.name
+
+  tags = merge({
+    ssh-user = var.user
+    Project  = var.project_name
+  }, var.tags)
+
+  cpu    = var.cpu
+  memory = "${var.memory}Gi"
+
+  efi         = var.efi
+  secure_boot = var.efi ? var.secure_boot : false
+
+  dynamic "network_interface" {
+    for_each = data.harvester_network.this
+    content {
+      name           = network_interface.value.name
+      network_name   = network_interface.value.id
+      type           = local.networks_map[network_interface.value.name].interface_type
+      model          = local.networks_map[network_interface.value.name].interface_model
+      wait_for_lease = local.networks_map[network_interface.value.name].wait_for_lease
+    }
+  }
+
+  dynamic "disk" {
+    for_each = local.disks_map
+    content {
+      name       = disk.value.name
+      type       = disk.value.type
+      size       = "${disk.value.size}Gi"
+      bus        = disk.value.bus
+      image      = index(var.disks, disk.value) == 0 ? (
+              var.image_name == null ? var.default_image_id : data.harvester_image.this[0].id
+      ) : null
+      boot_order = index(var.disks, disk.value) + 1 //boot_order starts at 1, while the index() function is 0-based
+      auto_delete = true
+    }
+  }
+
+  ssh_keys = compact([var.ssh_public_key_id, try(data.harvester_ssh_key.shared[0].id, null)])
+
+  # Default "USB Tablet" config for VNC usage
+  input {
+    name = "tablet"
+    type = "tablet"
+    bus  = "usb"
+  }
+
+  cloudinit {
+    user_data = local.cloud_init_user_data
+  }
+
+  // Allow for more than the default time for VM destruction
+  // Uncomment once https://github.com/harvester/harvester/issues/6647 is resolved
+  # timeouts {
+  #   delete = "15m"
+  # }
+}
+
+resource "null_resource" "cloud_init_wait" {
```

Comment:
You might want to use host_configuration_commands for this, so that there is only one mechanism that can be overridden if necessary

Reply:
Are you referring ot the usage of cloud-init user_data? I only kept the host_configuration_commands mechanism in place for parity with the other host modules, I could  remove that host_configuration block instead, as the user_data is the expected way to setup VMs.

If you're referring to the "cloud_init_wait" block, I can move that to the host_configuration_commands 👍

Reply:
> If you're referring to the "cloud_init_wait" block, I can move that to the host_configuration_commands 👍

Yeah, sorry for not being very clear with my comment.


Suggestion was to piggyback the existing `host_configuration_commands` mechanism, adding a command to the default list.

Thanks!

Reply:
This should be resolved